### DISCUSSION TITLE
Is there a way to pass payload/data to eventless transition?

### DISCUSSION THREAD

**Discussion by dsngo** [statelyai/xstate#1823] [Q&A] [answered]
I recently ran into a problem that my intermediate state contains eventless transition to reroute the state and it'd require additional data/payload from previous event/state for condition. I realized that eventless transition always set the event parameter to an xstate event object as `{type: ""}`, and there is no way to add payload to this event (which I assume it is an `eventless`, but should it payloadless as well?). My solution is to reference last event from `condMeta` state. I feel a bit hacky, is there other way to do this?

**Comment by Andarist** [comment, marked as answer]
This will change in the next major version - there wont be `''` event, you will have access to the last "real" event in your eventless transitions

**Reply by dsngo** [reply]
Great to know, thank @Andarist.
